src/gui/reports/ReportsWidgetPasskeys.cpp

```diff
@@ -0,0 +1,330 @@
+
+        const QList<QSharedPointer<Item>>& items() const
+        {
+            return m_items;
+        }
+
+    private:
+        QSharedPointer<Database> m_db;
+        QList<QSharedPointer<Item>> m_items;
+    };
+} // namespace
+
+PasskeyList::PasskeyList(const QSharedPointer<Database>& db)
+    : m_db(db)
+{
+    for (auto group : db->rootGroup()->groupsRecursive(true)) {
+        // Skip recycle bin
+        if (group->isRecycled()) {
+            continue;
+        }
+
+        for (auto entry : group->entries()) {
+            if (entry->isRecycled() || !entry->attributes()->hasKey(BrowserPasskeys::KPEX_PASSKEY_PRIVATE_KEY_PEM)) {
+                continue;
+            }
+
+            const auto item = QSharedPointer<Item>(new Item(group, entry));
+            m_items.append(item);
+        }
+    }
+}
+
+ReportsWidgetPasskeys::ReportsWidgetPasskeys(QWidget* parent)
+    : QWidget(parent)
+    , m_ui(new Ui::ReportsWidgetPasskeys())
+    , m_referencesModel(new QStandardItemModel(this))
+    , m_modelProxy(new QSortFilterProxyModel(this))
+{
+    m_ui->setupUi(this);
+
+    m_modelProxy->setSourceModel(m_referencesModel.data());
+    m_modelProxy->setSortLocaleAware(true);
+    m_ui->passkeysTableView->setModel(m_modelProxy.data());
+    m_ui->passkeysTableView->horizontalHeader()->setSectionResizeMode(QHeaderView::Interactive);
+    m_ui->passkeysTableView->verticalHeader()->setSectionResizeMode(QHeaderView::ResizeToContents);
+
+    connect(m_ui->passkeysTableView, SIGNAL(customContextMenuRequested(QPoint)), SLOT(customMenuRequested(QPoint)));
+    connect(m_ui->passkeysTableView, SIGNAL(doubleClicked(QModelIndex)), SLOT(emitEntryActivated(QModelIndex)));
+    connect(m_ui->passkeysTableView->selectionModel(),
+            SIGNAL(selectionChanged(QItemSelection, QItemSelection)),
+            this,
+            SLOT(selectionChanged()));
+    connect(m_ui->excludeExpired, SIGNAL(stateChanged(int)), this, SLOT(updateEntries()));
+    connect(m_ui->exportButton, SIGNAL(clicked(bool)), this, SLOT(exportPasskey()));
+    connect(m_ui->importButton, SIGNAL(clicked(bool)), this, SLOT(importPasskey()));
+
+    m_ui->exportButton->setEnabled(false);
+
+    new QShortcut(Qt::Key_Delete, this, SLOT(deleteSelectedEntries()));
+}
+
+ReportsWidgetPasskeys::~ReportsWidgetPasskeys()
+{
+}
+
+void ReportsWidgetPasskeys::addPasskeyRow(Group* group, Entry* entry, bool excluded)
+{
+    StateColorPalette statePalette;
+
+    auto urlList = entry->getAllUrls();
+    auto urlToolTip = tr("List of entry URLs");
+
+    auto title = entry->title();
+    if (excluded) {
+        title.append(tr(" (Excluded)"));
+    }
+
+    auto row = QList<QStandardItem*>();
+    row << new QStandardItem(Icons::entryIconPixmap(entry), title);
+    row << new QStandardItem(Icons::groupIconPixmap(group), group->hierarchy().join("/"));
+    row << new QStandardItem(entry->username());
+    row << new QStandardItem(urlList.join('\n'));
```

Comment:
This needs to show `KPEX_PASSKEY_RELYING_PARTY`

Reply:
I'll add it.